Tests/TensorFlowTests/OperatorTests/MathTests.swift

```diff
@@ -346,6 +346,14 @@ final class MathOperatorTests: XCTestCase {
         let ySmall = xSmall.logSumExp()
         let expectedYSmall = Tensor<Float>(-235.28029)
         assertEqual(ySmall, expectedYSmall, accuracy: 0.0001)
+
+        let xLarge = Tensor<Float>([
```

Comment:
Does this test case fail before you make the change? That would be surprising because it doesn't look like anything should be infinite in this test case. If it does really fail before you make the change, I'd suspect that there is some bug in one of the calls you removed in this PR, like `withoutDerivative(at:)`, `Tensor(zerosLike:)`, `.replacing`, or `.isFinite`. And if that is the case, then we should fix that bug instead of removing the finiteness check.

Reply:
This test case would fail if the `logSumExp` was implemented without subtracting the max first. I believe the instability in my use case comes from differentiating the current `logSumExp` implementation (not sure which part or why though). I will try to create a test case for that (it's a bit difficult because I only encounter the issue in while performing inference in a Bayesian model). In general though, I don't know why that finiteness check is necessary. The result should be the same if certain elements are non-finite irrespective of whether we have that check. I know it's there in the Python implementation but I can't really understand why it's necessary.